### PR TITLE
Limit checks for Composition resource and Sync Strategies to Android Client Role

### LINKED ISSUE #124: Issues encountered when configuring FHIR Web to work with the Gateway on a fresh install
The following issues related to the FHIR Gateway have been encountered while setting up a new user for FHIR Web Front End

- [ ] No Composition resource found for application id 
- [ ] User un-authorized to GET /Patient. User assignment or sync strategy not configured correctly.

The errors above point to there being a requirement  for having an existing `Composition` resource and user assignments even when setting up the initial user that can be used on the FHIR Web Front End

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Resolves #124 

**Engineer Checklist**

- [ ] I have written **Unit tests** for any new feature(s) and edge cases for
      bug fixes
- [ ] I have added documentation for any new feature(s) and configuration
      option(s) on the `README.md`
- [ ] I have run `mvn spotless:check` to check my code follows the project's
      style guide
- [ ] I have run `mvn clean test jacoco:report` to confirm the coverage report
      was generated at `plugins/target/site/jacoco/index.html`
- [ ] I ran `mvn clean package` right before creating this pull request.

### CODE REVIEW

File: plugins/src/main/java/org/smartregister/fhir/gateway/plugins/Utils.java [hunk 1]
```diff
@@ -301,4 +304,23 @@ public static String extractLogicalId(ResourceType resourceType, String url) {
...
+        if (matchedRoles.size() != 1) {
+            ForbiddenOperationException forbiddenOperationException =
+                    new ForbiddenOperationException(
+                            "User must have at least one and at most one of these client roles "
+                                    + Arrays.toString(Constants.CLIENT_ROLES));
+            ExceptionUtil.throwRuntimeExceptionAndLog(
+                    logger, forbiddenOperationException.getMessage(), forbiddenOperationException);
+        }
+        return matchedRoles.get(0);
+    }
```
Comment (dubdabasoduba) [inline plugins/src/main/java/org/smartregister/fhir/gateway/plugins/Utils.java:325]: Nice! moving this to the Utils

File: plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PermissionAccessChecker.java [hunk 2]
```diff
@@ -102,7 +103,15 @@ public AccessDecision checkAccess(RequestDetailsReader requestDetails) {
...
+
+        if (!clientRole.equals(Constants.ROLE_ANDROID_CLIENT)) {
+            syncStrategyIds = Collections.emptyMap();
+            this.syncAccessDecision =
+                    new SyncAccessDecision(
+                            fhirContext, jwt.getSubject(), true, syncStrategyIds, "", userRoles);
+            return;
+        }
         Composition composition = fetchComposition();
         String syncStrategy = readSyncStrategyFromComposition(composition);
```
Comment (dubdabasoduba) [inline plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PermissionAccessChecker.java:116]: @Rkareko This still seems to fetch composition.  This should only happen for users with the `ANDROID_CLIENT` role

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Review by pld — APPROVED** [review]
